src/algebraic_geometry/morphisms/quasi_separated.lean

```diff
@@ -274,4 +280,205 @@ begin
+    { fapply X.sheaf.eq_of_locally_eq' (λ i : s, i.1.1),
+      { refine λ i, hom_of_le _, erw hs, exact le_supr _ _ },
+      { exact le_of_eq hs },
+      { intro i,
+        replace hn := congr_arg (λ x, X.presheaf.map (hom_of_le
+          (le_trans (hs₁ i) le_sup_left)).op f ^ (finset.univ.sup n - n i) * x) (hn i),
+        dsimp only at hn,
+        delta Scheme.sheaf SheafedSpace.sheaf,
+        simp only [← map_pow, map_mul, ← comp_apply, ← functor.map_comp, ← op_comp, ← mul_assoc]
+          at hn ⊢,
+        erw [← map_mul, ← map_mul] at hn,
+        rw [← pow_add, ← pow_add, ← add_assoc, ← add_assoc, tsub_add_cancel_of_le] at hn,
+        convert hn,
+        exact finset.le_sup (finset.mem_univ _) } },
+    use finset.univ.sup n + n₁ + n₂,
+    -- By the sheaf condition, since `f ^ (n + n₂) * y₁ = f ^ (n + n₁) * y₂`, it can be glued into
+    -- the desired section on `S ∪ U`.
+    use (X.sheaf.obj_sup_iso_prod_eq_locus S U.1).inv ⟨⟨_ * _, _ * _⟩, this⟩,
+    fapply X.sheaf.eq_of_locally_eq₂,
+    rotate 5,
+    { exact X.basic_open (X.presheaf.map (hom_of_le le_sup_left).op f) },
+    { exact X.basic_open (X.presheaf.map (hom_of_le le_sup_right).op f) },
+    { refine hom_of_le _, rw X.basic_open_res, exact inf_le_right },
+    { refine hom_of_le _, rw X.basic_open_res, exact inf_le_right },
+    { rw [X.basic_open_res, X.basic_open_res],
+      erw ← inf_sup_right,
+      refine le_inf_iff.mpr ⟨X.basic_open_le f, le_of_eq rfl⟩ },
+    { convert congr_arg (X.presheaf.map (hom_of_le _).op)
+        (X.sheaf.obj_sup_iso_prod_eq_locus_inv_fst S U.1 ⟨⟨_ * _, _ * _⟩, this⟩) using 1,
+      swap 3,
+      { rw X.basic_open_res, exact inf_le_left },
+      { delta Scheme.sheaf SheafedSpace.sheaf,
+        simp only [← comp_apply (X.presheaf.map _) (X.presheaf.map _),
+          ← functor.map_comp, ← op_comp],
+        congr },
+      { delta Scheme.sheaf SheafedSpace.sheaf,
+        simp only [map_pow, map_mul, ← comp_apply, ← functor.map_comp, ← op_comp, mul_assoc,
+          pow_add], erw hy₁, congr' 1, rw [← mul_assoc, ← mul_assoc], congr' 1,
+        rw [mul_comm, ← comp_apply, ← functor.map_comp], congr } },
+    { convert congr_arg (X.presheaf.map (hom_of_le _).op)
+        (X.sheaf.obj_sup_iso_prod_eq_locus_inv_snd S U.1 ⟨⟨_ * _, _ * _⟩, this⟩) using 1,
+      swap 3,
+      { rw X.basic_open_res, exact inf_le_left },
+      { delta Scheme.sheaf SheafedSpace.sheaf,
+        simp only [← comp_apply (X.presheaf.map _) (X.presheaf.map _),
+          ← functor.map_comp, ← op_comp],
+        congr },
+      { delta Scheme.sheaf SheafedSpace.sheaf,
+        simp only [map_pow, map_mul, ← comp_apply, ← functor.map_comp, ← op_comp, mul_assoc,
+          pow_add], erw hy₂, rw [← comp_apply, ← functor.map_comp], congr } } }
+end
+
+/-- If `U` is qcqs, then `Γ(X, D(f)) ≃ Γ(X, U)_f` for every `f : Γ(X, U)`.
+This is known as the **Qcqs lemma** in Vakil's "The rising sea". -/
```

Comment:
Could we add "the rising sea" to the bibliography? I guess it keeps changing, but references into the literature are helpful for readers, both human and machine.